api/src/main/java/org/openmrs/module/kenyaemr/calculation/library/surveillance/EligibleForIDSRFlagsCalculation.java

```diff
@@ -0,0 +1,729 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.kenyaemr.calculation.library.surveillance;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.joda.time.DateTime;
+import org.joda.time.Days;
+import org.openmrs.*;
+import org.openmrs.api.ConceptService;
+import org.openmrs.api.PatientService;
+import org.openmrs.api.context.Context;
+import org.openmrs.calculation.patient.PatientCalculationContext;
+import org.openmrs.calculation.result.CalculationResultMap;
+import org.openmrs.module.kenyacore.calculation.*;
+import org.openmrs.module.kenyaemr.calculation.EmrCalculationUtils;
+import org.openmrs.module.kenyaemr.metadata.CommonMetadata;
+import org.openmrs.module.kenyaemr.metadata.HivMetadata;
+import org.openmrs.module.kenyaemr.util.EmrUtils;
+import org.openmrs.module.metadatadeploy.MetadataUtils;
+
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+/**
+ * Calculates the eligibility for ILI screening flag for  patients
+ *
+ * @should calculate Active visit
+ * @should calculate cough for <= 10 days
+ * @should calculate fever for <= 10 days
+ * @should calculate temperature  for >= 38.0 same day
+ * @should calculate duration < 10 days
+ * <p>
+ * SARI
+ * * @should calculate admitted
+ */
+public class EligibleForIDSRFlagsCalculation extends AbstractPatientCalculation implements PatientFlagCalculation {
+	protected static final Log log = LogFactory.getLog(EligibleForIDSRFlagsCalculation.class);
+
+	StringBuilder idsrMessage = new StringBuilder();
```

Comment:
So that you don't have to do a lot of string concatenation, you can check on using a list of string and then join the elements using a ','.